sentry/src/main/java/io/sentry/TransactionListener.java

```diff
@@ -0,0 +1,13 @@
+package io.sentry;
+
+import org.jetbrains.annotations.NotNull;
+
+public interface TransactionListener {
```

Comment:
`m`: This is new to me. Is this something we want to be public? Did we sync this with other SDKs?

Reply:
this is needed since we have a Java/Android SDK and transactions live only in the Java SDK but the Android SDK needs some tweaks, it was discussed yesterday in the daily standup

Reply:
Ah, sorry, I missed that. As we have `BeforeSendCallback` and `BeforeBreadcrumbCallback`, should we maybe name this `TransactionFinishedCallback` and the method below `execute`? @bruno-garcia, what do you think about the naming here?

Reply:
actually its not the same concept as `BeforeSendCallback` and `BeforeBreadcrumbCallback` which is called for every event/breadcrumb.
so I'd not like to add the `Callback` suffix as it could be misunderstood.
the `TransactionListener` is just a callback that is called only when using the `waitForChildren` feature (currently only on Android), so I can trigger some Android code before the transaction gets captured.

Reply:
> the TransactionListener is just a callback that is called

😁. No strong opinion here. For me, `TransactionListener` is a bit confusing.

Reply:
By looking at the usage:
![image](https://user-images.githubusercontent.com/1633368/126345434-3503d4a0-7c77-4a1e-ba53-b16cf5f38fda.png)

It seems to me it's a callback so something like `TransactionFinishedCallback` would make sense.

Reply:
I can rename, but the `Span` class takes a `SpanListener` which does pretty much the same thing, so ideally we'd align both of them introducing a breaking change, well its an internal class so not a big deal

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SpanListener.java

so do I rename both?

Reply:
If it's not a breaking change, then I would yes.

Reply:
done thanks